lib/bundlex/toolchain/common/unix/os_deps.ex

```diff
@@ -8,11 +8,27 @@ defmodule Bundlex.Toolchain.Common.Unix.OSDeps do
   def resolve_os_deps(native) do
     {cflags_list, libs_list} =
       native.os_deps
-      |> Enum.map(fn {provider_or_providers, lib_name_or_names} ->
-        providers = Bunch.listify(provider_or_providers)
-        lib_names = Bunch.listify(lib_name_or_names)
+      |> Enum.map(fn
+        {name, providers} when is_atom(name) and name != :pkg_config ->
```

Comment:
It's possible that I am overthinking that, but imagine somebody who uses new syntax and wants to group the dependencies resolved via pkgconfig as a single entry of os_deps:
```
os_deps: [
  ...
  pkg_config: {:pkgconfig, [<some pkgconfig names>]},
  ...
]
```
He would get warning completely unrelated to the origin of the problem. Perhaps it would be good to describe such a case in the warning message?